lib/src/test/java/com/cloudbees/groovy/cps/sandbox/SandboxInvokerTest.java

```diff
@@ -696,4 +696,16 @@ public void sandboxInterceptsImplicitCastsArrayAssignment() throws Throwable {
                 "Script1.method3()");
     }
 
+    @Test public void castsInTrustedCodeCalledByUntrustedCodeShouldBeTrusted() throws Throwable {
+        TrustedCpsCompiler trusted = new TrustedCpsCompiler();
+        trusted.setUp();
+        getBinding().setVariable("trusted", trusted.getCsh().parse("def foo() { File f = ['secret.key'] }"));
+
+        assertIntercept(
+            "trusted.foo()", // Untrusted script
+            new File("secret.key"),
+            "Script1.super(Script1).setBinding(Binding)",
+            "Script1.trusted",
+            "Script1.foo()");
```

Comment:
Previously, we would also intercept `new File(String)`, which was wrong because that is in trusted code that should not be subject to the sandbox.